proposals/0000-bytearray-literals.rst

```diff
@@ -0,0 +1,189 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+.. highlight:: haskell
+
+This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/N>`_.
+
+.. contents::
+
+ByteArray# literals
+===================
+
+This is a proposal to introduce ``ByteArray#`` literals. The user
+would be able to write:
+
+.. code-block:: haskell
+
+  "Literals"#         -- Addr# (Modified UTF-8)
```

Comment:
Let's get rid of `Modified UTF-8`, we really don't need one, do we?

Reply:
If we get rid of it, this won't be backwards compatible. I'm concerned about places that may rely on the existing behavior. The libraries `short-text` and `text` come to mind, but this change would only manifest itself in obscure corner cases, and it would be hard for users of GHC to diagnose it.

Reply:
Actually, I completely misunderstood the way GHC currently works. I thought that it used Modified UTF-8 for `Addr#` literals, but it doesn't. It only does that when desugaring `String` literals.